icu4c/source/i18n/reldatefmt.cpp

```diff
@@ -841,43 +863,122 @@ UDateRelativeDateTimeFormatterStyle RelativeDateTimeFormatter::getFormatStyle()
     return fStyle;
 }
 
+#define FORMAT_BOILERPLATE_START { \
```

Comment:
I understand you try to share code across different functions and use these macro, but I found it very difficult to follow. Could you instead call into a common function with a callback function which you can implemented as a C++ lambda to share code instead?
instead of 

A () {
 BEGIN
 a_unique_code
 END
}
B() {
 BEGIN
 b_unique_code
 END
}

Change it to
A() {
 common(... []->() { a_unique_code}))
}
B() {
 common(... []->() { b_unique_code}))
}

common( callback) {
 BEGIN
 callback()
 END
}

Reply:
Done.